firebase/test/core/data/gateways/firestore-gateways.spec.ts

```diff
@@ -108,5 +115,59 @@ describe("FirestoreGateway", () => {
         assert(transactionMock.get.calledOnce);
       });
     });
+
+    describe("getCollection", () => {
```

Comment:
NIT: This suggestion apply to all next test PR's: I think they could be split into separate files. Although they belong to the same "context" opening a 400+ lines file is not pleasant at all, even when things are nested as the test structures are.

Reply:
Looking at third-party open-sourced projects, that's pretty standard of unit testing, but please correct me if I'm wrong. I think that's up to your usage of the IDE's or even vim/emacs. This file is simply insanely verbose because of the amount of mocking/stubbing.

Slightly offtopic: if we argue about vscode, which is the standard that we use in memo, I've learned to navigate using symbols pretty decently, so I am completely fine with having files that have 300+ lines if they are coherent and have a clear responsibility, given the "architecture".

<details>
<summary>Embedded symbol navigation</summary>

![Screen Shot 2021-09-27 at 17 13 07](https://user-images.githubusercontent.com/13678134/134978399-3f02e467-6153-46e1-a5f1-cb329b99a140.png)

</details>

[If you press `CMD/CTRL + SHIFT + O`, you'll open the symbols navigation, which you can quickly filter/navigate to specific properties, functions and whatnot.
](https://code.visualstudio.com/Docs/editor/editingevolved#_go-to-symbol)

<details>
<summary>Filtered symbol navigation</summary>

![Screen Shot 2021-09-27 at 17 18 44](https://user-images.githubusercontent.com/13678134/134979111-65aeb7a0-c5cc-4f69-8f15-fbec9833ee7d.png)

</details>